uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/AbsenceType.php

```diff
@@ -540,4 +541,45 @@ public static function getOneWithMustTakePublicHolidayAsLeaveRequest() {
 
     return null;
   }
+
+  /**
+   * Returns the date the TOIL will expire based on the date given to the method.
+   *
+   * @param \DateTime $date
+   *   The date to calculate TOIL expiry date
+   *
+   * @return \DateTime|null
+   *   returns null when TOIL never expires for the Absence Type
+   *
+   * @throws CRM_HRLeaveAndAbsences_Exception_InvalidAbsenceTypeException
+   */
+  public function calculateToilExpiryDate(DateTime $date) {
+
+    if (!$this->allow_accruals_request) {
+      throw new InvalidAbsenceTypeException("This Absence Type does not allow Accruals Request");
+    }
+    if ($this->toilNeverExpires()) {
+      return null;
+    }
+
+    if ($this->accrual_expiration_unit == self::EXPIRATION_UNIT_DAYS) {
+      $unit = 'day';
+    }
+    if ($this->accrual_expiration_unit == self::EXPIRATION_UNIT_MONTHS) {
+      $unit = 'month';
+    }
+
+    $date->modify('+'.$this->accrual_expiration_duration. ' ' . $unit);
```

Comment:
Instead of modify the received date, it's safer to clone it and modify the clone:
```php
$expiryDate = clone $date;
$expiryDate->modify('+'.$this->accrual_expiration_duration. ' ' . $unit);
```
The callers of this method probably won't expected that the passed date will be changed.